static/src/javascripts/projects/common/modules/ui/relativedates.spec.js

```diff
@@ -0,0 +1,197 @@
+    },
+    moreThanFiveDaysAgo: {
+        date: '2012-08-05T21:30:00+01:00',
+        expectedOutput: '5 Aug 2012',
+        expectedShortOutput: '5 Aug 2012',
+        expectedMedOutput: '5 Aug 2012',
+        expectedLongOutput: '5 Aug 2012',
+    },
+    oneMinuteAgoInAnotherTimeZone: {
+        date: '2012-08-13T12:58:40+02:00',
+        expectedOutput: '1m',
+        expectedShortOutput: '1m',
+        expectedMedOutput: '1m ago',
+        expectedLongOutput: '1 minute ago',
+    },
+};
+
+Object.keys(datesToTest).forEach(category => {
+    describe(`Show relative dates for timestamps formatted as YYYY-MM-DD HH:MM:SS [${category}]`, () => {
+        const d = datesToTest[category];
+        const epoch = Date.parse(d.date);
+
+        test('standard', () => {
+            expect(makeRelativeDate(epoch)).toBe(d.expectedOutput);
+        });
+
+        test('short', () => {
+            expect(makeRelativeDate(epoch, { format: 'short' })).toBe(
+                d.expectedShortOutput
+            );
+        });
+
+        test('med', () => {
+            expect(makeRelativeDate(epoch, { format: 'med' })).toBe(
+                d.expectedMedOutput
+            );
+        });
+
+        test('long', () => {
+            expect(makeRelativeDate(epoch, { format: 'long' })).toBe(
+                d.expectedLongOutput
+            );
+        });
+    });
+});
+
+test('Return the input date if said date is in the future', () => {
+    const epochBug = '2038-01-19T03:14:07';
+    expect(makeRelativeDate(Date.parse(epochBug))).toBe(false);
+});
+
+test('Fail politely if given non-date / invalid input for either argument', () => {
+    const nonsensicalEpoch = 10000000000000; // Beyond valid date range
+    expect(makeRelativeDate(nonsensicalEpoch)).toBe(false);
+});
+
+test("Fail politely if the date is older than a 'notAfter' value", () => {
+    const tenSecondsAgo = new Date(fakeNow - 10 * 1000);
+    expect(makeRelativeDate(tenSecondsAgo, { notAfter: 9 })).toBe(false);
+});
+
+test('Convert valid timestamps in the HTML document into their expected output', () => {
+    relativeDatesInit();
+    const el = document.getElementById('time-valid');
+    const text = el ? el.innerHTML : '';
+    const title = el ? el.getAttribute('title') : '';
+
+    expect(text).toBe('Yesterday 19:43');
+    expect(title).toBe('12th August');
+});
+
+test('Ignore invalid timestamps', () => {
+    relativeDatesInit();
+    const el = document.getElementById('time-invalid');
+    const text = el ? el.innerHTML : '';
+    expect(text).toBe('Last Tuesday');
+});
+
+test('Should convert timestamps in document to users locale', () => {
+    replaceLocaleTimestamps();
+    const el = document.getElementById('time-locale');
+    const text = el ? el.innerHTML : '';
+    expect(text).toBe('17:00');
+});
+
+test('Should convert timestamps in HTML element to users locale', () => {
+    // TODO remove - not sure of value of this test?
```

Comment:
This test ensures the timestamp is updated and added to the passed element. Do you think it doesn't add value given the fact we're already testing the function without passing an element?

Reply:
I just misread it, as it is indeed testing different things (one passes in an elem, the other does not and checks the body is modified). So have reimplemented.